impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -482,17 +502,21 @@ public boolean remove(final K key, final V value) throws CacheAccessException {
       if (removedValue.get() != null) {
         conditionalRemoveObserver.end(StoreOperationOutcomes.ConditionalRemoveOutcome.REMOVED);
         decrementCurrentUsageInBytesIfRequired(removedValue.get().size());
-        return true;
+        return RemoveStatus.REMOVED;
       } else {
         conditionalRemoveObserver.end(StoreOperationOutcomes.ConditionalRemoveOutcome.MISS);
```

Comment:
Since we are anyway tracking the different between key present and key missing, I would also update the `conditionalRemoveObserver` to take a `RemoveStatus`, simplifying the code here.

This may apply to others methods modified in a similar way.